lib/galaxy/metadata/set_metadata.py

```diff
@@ -270,6 +274,13 @@ def set_meta(new_dataset_instance, file_dict):
         set_meta_kwds = stringify_dictionary_keys(json.load(open(filename_kwds)))  # load kwds; need to ensure our keywords are not unicode
         try:
             external_filename = unnamed_id_to_path.get(dataset_instance_id, dataset_filename_override)
+            if not os.path.exists(external_filename):
+                matches = glob.glob(external_filename)
+                assert len(matches) == 1, f"More than one file matched by output glob '{external_filename}'"
+                external_filename = matches[0]
+                assert safe_contains(tool_job_working_directory, external_filename), f"Cannot collect output '{external_filename}' from outside of working directory"
+                created_from_basename = os.path.relpath(external_filename, os.path.join(tool_job_working_directory, 'working'))
+                dataset.dataset.created_from_basename = created_from_basename
```

Comment:
I know you ... don't love this @jmchilton, but at least we preserve `created_from_basename` in this route.